test/e2e/e2e_test.go

```diff
@@ -0,0 +1,106 @@
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/engine"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/runtime"
+)
+
+// These tests confirm that all container and operator policies properly pass
+// with a known-good image, and properly fail with a known-bad image.
+// Any check that is found in the error section of the Result will cause this
+// to fail.
+var _ = Describe("policy validation", func() {
+	Describe("When enforcing operator policy", func() {
+		var (
+			// TODO: replace this with a project-accessible namespace once identified.
+			goodImage = "quay.io/komish/preflight-test-bundle-passes:latest"
+			badImage  = "quay.io/komish/preflight-test-bundle-fails:latest"
+		)
+
+		Context("with a known-good image", func() {
+
+			cfg := runtime.Config{
+				Image:         goodImage,
+				EnabledChecks: engine.OperatorPolicy(),
+			}
+
+			engine, err := engine.NewForConfig(cfg)
+			Expect(err).To(BeNil())
+
+			engine.ExecuteChecks()
+			results := engine.Results()
+
+			It("should pass all checks", func() {
+				Expect(len(results.Passed)).To(Equal(len(cfg.EnabledChecks)))
+			})
+		})
+
+		Context("with a known-bad image", func() {
+			cfg := runtime.Config{
+				Image:         badImage,
+				EnabledChecks: engine.OperatorPolicy(),
+			}
+
+			engine, err := engine.NewForConfig(cfg)
+			Expect(err).To(BeNil())
+
+			engine.ExecuteChecks()
+			results := engine.Results()
+
+			// TODO: Replace this check so that you test for individual check failures
+			It("should not pass any checks", func() {
+				Expect(len(results.Passed)).To(BeZero())
```

Comment:
Probably ensuring that the length of passed tests are less than number of enabled checks is a bit more generic?

Reply:
What would be the goal here?

Reply:
The goal is to have a more generic assertion; for example if we add a test image in the future where only one check fails and the remaining 10 checks pass, this test still fails.

Reply:
Ah, understod. I would expect this test to always fail. When we add images where a single check might fail, then I would expect a corresponding `Context` block, and a unique `It` block for that case. I believe this check is correct as written until we start building images that check for specific failures.

Reply:
Fair enough :)